lib/ansible/cli/doc.py

```diff
@@ -1339,6 +1388,20 @@ def get_role_man_text(self, role, role_json):
                     text.append(DocCLI._indent_lines(DocCLI._dump_yaml(doc['attributes'][k]), opt_indent))
                 del doc['attributes']
 
+            if doc.get('notes', False):
+                text.append("")
+                text.append(_format("NOTES:", 'bold'))
+                for note in doc['notes']:
+                    text.append(DocCLI.warp_fill(DocCLI.tty_ify(note), limit - 6,
+                                                 initial_indent=opt_indent[:-2] + "* ", subsequent_indent=opt_indent))
+                del doc['notes']
+
+            if doc.get('seealso', False):
+                text.append("")
+                text.append(_format("SEE ALSO:", 'bold'))
+                DocCLI._add_seealso(text, doc['seealso'], limit=limit, opt_indent=opt_indent)
+                del doc['seealso']
```

Comment:
This can be made less repetitive by using `doc.pop` and the walrus operator. Then there's only one reference to the key instead of three.

For example:

```python
        if notes := doc.pop('notes', None):
```

The same technique can be applied to the two other new conditionals. There's no need to update any of the existing code, just the portions new to this PR.